Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs

```diff
@@ -62,25 +92,89 @@ internal void InstallOrUpdate(UserPlugin plugin)
+                          CurrentVersion = existingPlugin.Metadata.Version, 
+                          NewVersion = pluginFromManifest.Version,
+                          existingPlugin.Metadata.IcoPath,
+                          PluginExistingMetadata = existingPlugin.Metadata,
+                          PluginNewUserPlugin = pluginFromManifest
+                      };
+
+            if (!resultsForUpdate.Any())
+                return new List<Result> { 
+                                new Result 
+                                    {
+                                        Title = Context.API.GetTranslation("plugin_pluginsmanager_update_noresult_title"),
+                                        SubTitle = Context.API.GetTranslation("plugin_pluginsmanager_update_noresult_subtitle"),
+                                        IcoPath = icoPath
+                                    }};
+
+
+            var results = resultsForUpdate
+                            .Select(x =>
+                                new Result
+                                    {
+                                        Title = $"{x.Name} by {x.Author}",
+                                        SubTitle = $"Update from version {x.CurrentVersion} to {x.NewVersion}",
+                                        IcoPath = x.IcoPath,
+                                        Action = e =>
+                                        {
+                                            string message = string.Format(Context.API.GetTranslation("plugin_pluginsmanager_update_prompt"),
+                                                                        x.Name, x.Author,
+                                                                        Environment.NewLine, Environment.NewLine);
+
+                                            if (MessageBox.Show(message, Context.API.GetTranslation("plugin_pluginsmanager_update_title"),
+                                                                                                        MessageBoxButton.YesNo) == MessageBoxResult.Yes)
+                                            {
+                                                Uninstall(x.PluginExistingMetadata);
+
+                                                var downloadToFilePath = Path.Combine(DataLocation.PluginsDirectory, $"{x.Name}-{x.NewVersion}.zip");
+                                                Http.Download(x.PluginNewUserPlugin.UrlDownload, downloadToFilePath);
+                                                Install(x.PluginNewUserPlugin, downloadToFilePath);
+
+                                                Context.API.RestartApp();
+
+                                                return true;
+                                            }
+
+                                            return false;
+                                        }
+                                    })
+                            .ToList();
```

Comment:
Actaully, we can remove the ToList() here to keep it as an IEnumerable to make Linq calculate the where and select method together, which should be faster (though not significant), since this result is only calculated once else where.